rfcs/20250520-async-threadpool-support/README.md

```diff
@@ -0,0 +1,157 @@
+# Proposal to support asynchronous threadpool runtime
+
+## Background
+
+XLA is switching to a new threapool runtime on CPU. In particular:
+- it has an asynchronous `parallel_for` implementation, meaning the
+  `parallel_for` function call returns while submitted tasks are still
+  running on the threadpool.
+- XLA will have a single threadpool for both primitive creation and
+  execution. This allows to overlap primitive creation time with
+  primitive execution. As a result, multiple primitives will be
+  submitted from a threadpool to that same threadpool, so parallelism
+  will always be nested.
+
+We have some properties that we can leverage on oneDNN side:
+- Primitive and memory object lifetimes will be maintained by thunk
+  runtime while execution happens: no need to reference count oneDNN
+  objects to extend their lifetime.
+- consecutive `parallel_for` calls happen in order (similar to sycl
```

Comment:
In ACL, we usually chain calls to standalone post op calls after the main conv/ip call - for cases where we dont support explicit fusion https://github.com/uxlfoundation/oneDNN/blob/73ec3de65693929c1089ce225b37dff6bd71dffa/src/cpu/aarch64/acl_post_ops.cpp Wondering what impact it would have such a case?

Reply:
Good question, the answer is it depends on the dependency between these calls.
- if internally, ACL takes the threadpool as input and does no computation on input/output data on the main thread (all actual computations go through the threadpool), then no problem with keeping these calls on the main thread.
- if internally, ACL does some computation on input/output in the main thread, then these should be submitted to the threadpool so that they are executed in-order with the previous call. I would suggest to make the submission inside ACL though, and not on oneDNN side. This allow to safely call ACL API from main thread, without ACL users guessing internal implementation details.